### PR TITLE
fix: `update:model-value` not triggering on add and remove

### PR DESCRIPTION
**Pull request by NicolaiHorlacher**
When `onAdd` and `onRemoved` is triggered the list got updated in-place.
This leads to an undesired prop manipulation in the `VueDraggable` component, because the writable computed that manages the `update:model-value` emit isn't triggered trough in-place updates of arrays.

The `onUpdate` method already had a fix to sidestep this behavior. So, I applied it to `onAdd` and `onRemoved` also.

While testing I noticed that the `onRemove` method is reverting the SortableJS DOM manipulation only when clone mode is active. SortableJS removes the DOM element in any case, so `insertNodeAt` needs to be called outside of the if-statement.

### In conclusion:
These changes streamline the behavior of the composable with writable computeds and fixes missing `update:model-value` emits.
By extension it's now possible to use the component with `:model-value` when automatic updates of the list are not desired.

### CODE REVIEW

**Comment by Alfred-Skyblue** [comment]
Thanks!

**Comment by Alfred-Skyblue** [comment]
close #138

**Comment by Alfred-Skyblue** [comment]
close #77
